homu/server.py

```diff
@@ -558,19 +559,20 @@ def report_build_res(succ, url, builder, state, logger, repo_cfg):
     if succ:
         if all(x['res'] for x in state.build_res.values()):
             state.set_status('success')
-            desc = 'Test successful'
-            utils.github_create_status(state.get_repo(), state.head_sha,
-                                       'success', url, desc, context='homu')
-
-            urls = ', '.join('[{}]({})'.format(builder, x['url']) for builder, x in sorted(state.build_res.items()))  # noqa
-            test_comment = ':sunny: {} - {}'.format(desc, urls)
+            utils.github_create_status(
+                state.get_repo(), state.head_sha,
+                'success', url, "Test successful", context='homu'
+            )
 
             if state.approved_by and not state.try_:
-                comment = (test_comment + '\n' +
-                           'Approved by: {}\nPushing {} to {}...'
-                           ).format(state.approved_by, state.merge_sha,
-                                    state.base_ref)
-                state.add_comment(comment)
+                state.add_comment(comments.BuildCompleted(
+                    approved_by=state.approved_by,
+                    base_ref=state.base_ref,
+                    builders=dict(
+                        (k, v["url"]) for k, v in state.build_res.items()
+                    ),
```

Comment:
I'm quite sure we have a sufficiently new Python that supports dictionary comprehension...

```python
{k: v['url'] for k, v in build_res.items()}
```